tests/contrib/sysctl/test_watermark_scale_factor.py

```diff
@@ -0,0 +1,619 @@
+#!/usr/bin/env python
+
+from charmhelpers.contrib.sysctl.watermark_scale_factor import watermark_scale_factor, calculate_watermark_scale_factor, get_memtotal, get_normal_managed_pages
+
+from mock import patch, mock_open
+import unittest
+
+TO_PATCH = [
+    "log",
+    "ERROR",
+    "DEBUG"
+]
+
+PROC_MEMINFO = """
+MemTotal:       98881012 kB
+MemFree:         5415708 kB
+MemAvailable:   30993024 kB
+Buffers:           12712 kB
+Cached:         16951140 kB
+SwapCached:       155104 kB
+Active:         39450516 kB
+Inactive:       16879984 kB
+Active(anon):   25640412 kB
+Inactive(anon): 12953116 kB
+Active(file):   13810104 kB
+Inactive(file):  3926868 kB
+Unevictable:      156348 kB
+Mlocked:          156348 kB
+SwapTotal:       2097148 kB
+SwapFree:         702676 kB
+Dirty:              1708 kB
+Writeback:            12 kB
+AnonPages:      39295640 kB
+Mapped:          2094992 kB
+Shmem:            141668 kB
+KReclaimable:    8832856 kB
+Slab:           32299340 kB
+SReclaimable:    8832856 kB
+SUnreclaim:     23466484 kB
+KernelStack:      132608 kB
+PageTables:       229164 kB
+NFS_Unstable:          0 kB
+Bounce:                0 kB
+WritebackTmp:          0 kB
+CommitLimit:    51537652 kB
+Committed_AS:   112598920 kB
+VmallocTotal:   34359738367 kB
+VmallocUsed:     2191060 kB
+VmallocChunk:          0 kB
+Percpu:           393984 kB
+HardwareCorrupted:     0 kB
+AnonHugePages:   1744896 kB
+ShmemHugePages:        0 kB
+ShmemPmdMapped:        0 kB
+FileHugePages:         0 kB
+FilePmdMapped:         0 kB
+HugePages_Total:       0
+HugePages_Free:        0
+HugePages_Rsvd:        0
+HugePages_Surp:        0
+Hugepagesize:       2048 kB
+Hugetlb:               0 kB
+DirectMap4k:    85061760 kB
+DirectMap2M:    14542848 kB
+DirectMap1G:     2097152 kB
+"""
+
+PROC_ZONEINFO = """
```

Comment:
Although it is excellent to use real data, it might be appropriate to cut this down a bit please?